state-chain/pallets/cf-elections/src/vote_storage/nonce.rs

```diff
@@ -0,0 +1,82 @@
+use codec::{Decode, Encode};
+use frame_support::{
+	pallet_prelude::{Member, TypeInfo},
+	Parameter,
+};
+
+use super::{AuthorityVote, VoteComponents, VoteStorage};
+
+use crate::{CorruptStorageError, SharedDataHash};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, TypeInfo)]
+pub struct NonceVote<Value, Slot> {
```

Comment:
I think we can generalise it when it necessary? else it's a little confusing it has a general name and is used for one (quite specific) thing

Reply:
My view is that if we implement something generically, then it should have generic names. Reading generic code with specifically-named values is confusing because you always have the specific use-case in mind.